technique/muon/zero_field_setup.py

```diff
@@ -0,0 +1,278 @@
+            time.sleep(1)
+
+        # calculating variance of each field
+        var_x = np.var(fields_x)
+        var_y = np.var(fields_y)
+        var_z = np.var(fields_z)
+
+        rms = np.sqrt((var_x + var_y + var_z))
+
+        if plot is True:
+            time_in_seconds = [x for x in range(20)]
+            plt.figure()
+            plt.plot(time_in_seconds, fields_x, color="red", label="field X", marker="o")
+            plt.plot(time_in_seconds, fields_y, color="green", label="field Y", marker="o")
+            plt.plot(time_in_seconds, fields_z, color="blue", label="field Z", marker="o")
+            plt.xticks(time_in_seconds)
+            plt.xlabel("Time (sec)")
+            plt.ylabel("Field (mG)")
+            plt.legend(loc="upper left")
+            plt.show()
+
+        return rms
+
+    def close_plot(self):
+        """
+        close all plot
+        :return: none
+        """
+        plt.close("all")
+
+    def check_if_stray_field_exist(self):
+        """
+        Check if there is huge stray field
+        :return: True if there is huge stray field
+        """
+        measured_x, measured_y, measured_z = self.read_values()
+        if (abs(measured_x) < 4000 and abs(measured_y)
+                < 4000 and abs(measured_z) < 4000):
+            return False
+        return True
+
+    def get_single_corrected_field_value(self, field_direction):
+        """
+        get single corrected field value
+        :param field_direction: X, Y or Z
+        :return: corrected field value from the given direction
+        """
+        return g.get_pv(
+            "TE:NDLT1205:ZFMAGFLD_01:{}:CORRECTEDFIELD".format(field_direction))
+
+    def ask_user_to_continue(self):
+        """
+        Ask user if they want to continue further
+        :return:
+        """
+        reply = ""
+        while reply != "Y" and reply != "N":
+            reply = input("Would you like to continue? Y/N\n")
+        if reply == "N":
+            return "N"
+
+    def get_instrument_name(self):
+        """
+        get instrument name
+        :return: instrument name
+        """
+        return g.get_instrument()
+
+
+def update_offset():
+    """
+    updates the offset so that the corrected field reading is 0 for each axis
+    :return: None
+    """
+    x_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:X:_RAW")
+    y_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Y:_RAW")
+    z_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Z:_RAW")
+
+    x_new_offset = x_raw_measurement - 0
+    y_new_offset = y_raw_measurement - 0
+    z_new_offset = z_raw_measurement - 0
+
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:X:OFFSET", x_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Y:OFFSET", y_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Z:OFFSET", z_new_offset)
+
+
+def run_all():
+    """
+    Run all the functions in order for set up
+    :return:
+    """
+    while True:
```

Comment:
So that we can use `break` to break out of the loop without running the rest of the code if user wants to discontinue.

Reply:
That's quite a confusing pattern because it has the potential to repeat forever if you don't break. It would be better to use return statements I think